src/NHibernate/Dialect/DB2Dialect.cs

```diff
@@ -232,28 +232,29 @@ public override bool SupportsVariableLimit
 			get { return false; }
 		}
 
-		public override SqlString GetLimitString(SqlString querySqlString, SqlString offset, SqlString limit)
+		public override SqlString GetLimitString(SqlString sql, SqlString offset, SqlString limit)
 		{
 			if (offset == null)
 			{
-				return new SqlString(querySqlString,
-									 " fetch first ",
-									 limit,
-									 " rows only");
+				return new SqlString(sql, " fetch first ", limit, " rows only");
 			}
 
+			ExtractColumnOrAliasNames(sql, out var selectColumns, out _, out _);
+
 			/*
 			 * "select * from (select row_number() over(orderby_clause) as rownum, "
 			 * querySqlString_without select
 			 * " ) as tempresult where rownum between ? and ?"
 			 */
-			string rownumClause = GetRowNumber(querySqlString);
+			string rownumClause = GetRowNumber(sql);
 
 			SqlStringBuilder pagingBuilder = new SqlStringBuilder();
 			pagingBuilder
-				.Add("select * from (select ")
+				.Add("select " )
+				.Add(string.Join(",", selectColumns))
```

Comment:
```suggestion
				.Add(selectColumns)
```

Reply:
There is no such overload.

Reply:
It would have been a string with the previous suggested change, if that previous suggestion had been valid.